lib/pub_sub/poller.rb

```diff
@@ -1,31 +1,38 @@
 module PubSub
   class Poller
-    def initialize(queue_url, verbose = false)
-      @queue_url = queue_url
+    def initialize(verbose = false)
       @verbose = verbose
     end
 
     def poll
-      poller.poll(config) do |message|
-        if @verbose
-          PubSub.logger.info(
-            "PubSub received: #{message.message_id} - #{message.body}"
-          )
+      loop do
+        Breaker.current_breaker.run do
+          poller.poll(idle_timeout: 60) do |message|
+            if @verbose
+              PubSub.logger.info(
+                "PubSub received: #{message.message_id} - #{message.body}"
+              )
+            end
+            Message.new(message.body).process
+          end
         end
-        Message.new(message.body).process
+        Breaker.use_next_breaker
```

Comment:
The switch region is for the happy path. We don't know if the topic has gone down in us-east (what if we don't send topic messages?) , so it could be that another service is sending messages to us-west where we think us-east is fine.

To guard against this, we watch each region in turn, so messages sent to any region will be processed.